App.tsx

```diff
@@ -1,106 +1,22 @@
-import React, { useState } from 'react';
-import { StatusBar } from 'expo-status-bar';
-import { StyleSheet, View, FlatList, Image, TouchableOpacity, TextInput } from 'react-native';
-import axios from 'axios';
-import { Ionicons } from '@expo/vector-icons';
-import ImageItem from './ImageItem';
+import React from 'react';
+import { NavigationContainer } from '@react-navigation/native';
+import { createStackNavigator, StackScreenProps } from '@react-navigation/stack';
+import PhotoGalleryScreen from './PhotoGalleryScreen';
+import PhotoDetailScreen from './PhotoDetailScreen';
+import ImageModalScreen from './ImageModalScreen';
 
-interface ImageData {
-  id: number;
-  url: string;
-}
-
-const imageData: ImageData[] = [];
-for (let i = 1; i < 70; i++) {
-  imageData.push({ id: i, url: `https://picsum.photos/id/${i}/200` });
-}
-
-export default function App() {
-  const [selectedImage, setSelectedImage] = useState<string | null>(null);
-  const [searchTerm, setSearchTerm] = useState<string>('');
-
-  const handleSearch = (text: string) => {
-    setSearchTerm(text);
-  };
-
-  const handleImageSelect = (imageUrl: string) => {
-    setSelectedImage(imageUrl);
-  };
-
-  const handleImageClose = () => {
-    setSelectedImage(null);
-  };
-
-  const renderItem = ({ item }: { item: ImageData }) => (
-    <ImageItem url={item.url} onSelect={handleImageSelect} />
-  );
-
-  const filteredImages = imageData.filter((image) => image.id.toString().includes(searchTerm));
+const Stack = createStackNavigator();
 
+const App = () => {
   return (
-    <View style={styles.container}>
-      <TextInput
-        style={styles.searchInput}
-        placeholder="Search by ID..."
-        // I know this isn't required but I feel like restricting
-        // the user to only put in numbers would be acceptable when they're searching by number
-        keyboardType="numeric" 
-        value={searchTerm}
-        onChangeText={handleSearch}
-
-      />
-      {selectedImage ? (
-        <TouchableOpacity style={styles.modalImageContainer} onPress={handleImageClose}>
-          <Image source={{ uri: selectedImage }} style={styles.modalImage} />
-          <Ionicons name="close-circle-outline" size={36} color="white" style={styles.closeIcon} />
-        </TouchableOpacity>
-      ) : (
-        <FlatList
-          data={filteredImages}
-          renderItem={renderItem}
-          numColumns={3}
-          keyExtractor={(item) => item.id.toString()}
-        />
-      )}
-      <StatusBar style="auto" />
-    </View>
+    <NavigationContainer>
+      <Stack.Navigator initialRouteName="PhotoGallery">
+        <Stack.Screen name="PhotoGallery" component={PhotoGalleryScreen} />
+        <Stack.Screen name="PhotoDetail" component={PhotoDetailScreen} />
+        <Stack.Screen name="ImageModal" options={{headerShown: false}} component={ImageModalScreen} />
```

Comment:
Not quite what I was looking for here. We covered how to create a modal using React Navigation.
`<Stack.Screen name="ImageModal" options={{presentationStyle: 'modal'}} component={ImageModalScreen} />`
We don't want to hide the header either because we get a close button for free. I wanted the header customized so it was all black with a white close button.